0003-configuration-management-extension/README.md

```diff
@@ -0,0 +1,93 @@
+---
+name: Configuration Management Extension
+shortname: 3/CMX
+status: raw
+editor: Dmitry Sergeev <dsergeev@cybervisiontech.com>
+---
+
+## Introduction
+
+The Configuration Management Extension is a [Kaa Protocol](/0002-kaa-protocol/README.md) extension.
+
+It is intended to manage endpoint configuration distribution.
+
+## Requirements and constraints
+### Problems and possible solutions
+
+1. _Multiple configurations available for client._ It is possible that there will be more than one available configurations for endpoint (eg.: endpoint was offline while new configurations were added).
+   Solution:
+   - Only latest configuration should be sent to an endpoint to reduce amount of traffic.
+_Note:_ Approach for FOTA should be different.
+
+2. _The server should know if a configuration has been delivered._ 
+
+   Solutions:
+   - Use QoS levels for message delivery confirmation.
+
+     MQTT requires all PUBACK messages appear in the same order as received PUBLISH messages.
+
+3. _Configuration management extension obtains configuration from Endpoint Configuration Data Provider extensions_
+
+4. _Endpoint lifecycle events._ CMX should monitor endpoint lifecycle and deliver configuration updates if present.
+
+   Solutions:
+   - Integrate CMX with NATS endpoint lifecycle topic. If there's undelivered configuration updates, then send it when endpoint connects. Also, provide API for endpoint to request latest configuration. 
+
+5. _A device may have no ability to generate timestamps._
+
+   Solutions:
+   - On the server side, use network packet arrival time as a timestamp.
+   
+6. _Override configuration._ CMX should be able to work with different CDP extensions resulting. If there are different configuration records in multiple CDPs for one endpoint, then merge algorithm should be applied.
+   Solutions:
+   - TBD
+
+## Use cases
+
+### UC1
+Configuration delivery by request. The endpoint should be able to receive latest configuration from CMX by request.
+
+### UC2
+Configuration delivery as a reaction on EP lifecycle event. The endpoint should receive latest configuration when it connects to a server if this configuration hadn't applied yet.
+
+## Design
+
+### Request/response
+The extension uses request/response pattern. A response from client is sent to confirm delivery.
+
+For MQTT, responses MUST be published at `<request_path>/status`. Each response MUST be published with the same QoS as the corresponding request.
+
+### Formats
+#### Schemeless JSON
```

Comment:
The section is copy-pasted from DCE. It contains references to "data points" which is no relevant for configuration extension.